core/unit_test/TestViewAPI_e.hpp

```diff
@@ -120,12 +127,57 @@ void test_right_stride(Extents... extents) {
     ASSERT_EQ(all_strides[i], expected_stride);
     expected_stride *= view.extent(i);
   }
+  ASSERT_EQ(all_strides[view_type::rank()], expected_stride);
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  for (size_t i = view_type::rank(); i < size_t(8); ++i) {
+    ASSERT_EQ(view.stride(i), size_t(1));
+  }
+#endif
+}
+
+template <typename DataType, typename... Extents>
+void test_stride_stride(Extents... extents) {
+  using view_type =
+      Kokkos::View<DataType, Kokkos::LayoutStride, Kokkos::HostSpace>;
+
+  auto test = [](auto view_org) {
+    view_type view(view_org);
+
+    size_t all_strides[view_type::rank() + 1];
+    view.stride(all_strides);
+
+    size_t max_stride     = 0;
+    size_t max_stride_idx = 0;
+    for (size_t i = 0; i < view_type::rank(); ++i) {
+      ASSERT_EQ(view.stride(i), view_org.stride(i));
+      ASSERT_EQ(all_strides[i], view_org.stride(i));
+      if (view.stride(i) > max_stride) {
+        max_stride     = view.stride(i);
+        max_stride_idx = i;
+      }
+    }
+    ASSERT_EQ(all_strides[view_type::rank()],
+              max_stride * view.extent(max_stride_idx));
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+    for (size_t i = view_type::rank(); i < size_t(8); ++i) {
+      ASSERT_EQ(view.stride(i), size_t(0));
+    }
+#endif
+  };
+
+  Kokkos::View<DataType, Kokkos::LayoutRight, Kokkos::HostSpace> view_right(
+      "view", extents...);
+  test(view_right);
+  Kokkos::View<DataType, Kokkos::LayoutRight, Kokkos::HostSpace> view_left(
+      "view", extents...);
+  test(view_left);
```

Comment:
hmmm can we not compare the strides we get with a strided layout with the ones from the original view as a test?

Reply:
sry there is a "not" that should not be there. 
I was wondering if we actually add meaningful testing by doing the comparison. But I guess that is the best we can do for now